crates/uv-resolver/src/lock/target.rs

```diff
@@ -0,0 +1,278 @@
+use std::collections::{BTreeMap, VecDeque};
+
+use either::Either;
+use rustc_hash::FxHashSet;
+use uv_configuration::{BuildOptions, DevGroupsManifest, ExtrasSpecification, InstallOptions};
+use uv_distribution_types::{Resolution, ResolvedDist};
+use uv_normalize::{ExtraName, GroupName, PackageName, DEV_DEPENDENCIES};
+use uv_platform_tags::Tags;
+use uv_pypi_types::{ResolverMarkerEnvironment, VerbatimParsedUrl};
+use uv_workspace::dependency_groups::{DependencyGroupError, FlatDependencyGroups};
+use uv_workspace::Workspace;
+
+use crate::lock::{LockErrorKind, Package, TagPolicy};
+use crate::{Lock, LockError};
+
+/// A target that can be installed from a lockfile.
+#[derive(Debug, Copy, Clone)]
+pub enum InstallTarget<'env> {
```

Comment:
The core change here is that we used to have `InstallTarget` defined in `uv-workspace`, and none of the variants had `Lock`. Now, the `InstallTarget` has `Lock`, and the `to_resolution` method from `Lock` got moved over here. The benefit is that we can determine the workspace members without relying on discovering the `pyproject.toml` files of each member -- we can just read them from the lockfile, which enables `--frozen --all`.